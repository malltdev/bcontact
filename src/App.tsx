import styled, { css } from "styled-components";
import { useEffect, useState } from "react";
import { AppProvider } from "./context/AppProvider";
import Button from "./components/Button";
import ContactList from "./components/ContactList";
import GlobalStyles from "./utils/GlobalStyles";

export interface PropsContainer {
  align: boolean;
}

const Container = styled.div<PropsContainer>`
  position: fixed;
  bottom: 20px;
  z-index: 1000;
  ${(props) =>
    props.align
      ? css`
          left: 20px;
          & > div {
            justify-content: flex-start;
          }
        `
      : css`
          right: 20px;
          & > div {
            justify-content: flex-end;
          }
        `}
`;

function App() {
  const [align, setAlign] = useState(false);

  useEffect(() => {
    const rootElement = document.getElementById("BContact");
    const dataAlign = rootElement?.getAttribute("data-align") || "";
    const dataAlignLeft = dataAlign === "left";
    if (dataAlignLeft) {
      setAlign(true);
    }
  }, []);

  return (
    <AppProvider>
      <GlobalStyles />
      <Container align={align}>
        <ContactList />
        <Button />
      </Container>
    </AppProvider>
  );
}

export default App;

// REACT SHADOW https://codesandbox.io/s/by6bo?file=/src/index.js:223-232
