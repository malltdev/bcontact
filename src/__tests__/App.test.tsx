import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "../App";

// deve retornar um botÃ£o
test("should return a button", () => {
  render(<App />);
  const buttonElement = screen.getByTestId("button-bcontact");
  expect(buttonElement).toBeInTheDocument();
});

// deve retornar ao clicar em um container com a lista de contatos
test("should return when clicking a container with the list of contacts", async () => {
  const BContact = document.createElement("div");
  BContact.setAttribute("id", "BContact");
  BContact.setAttribute("data-color", "#2ea6d9");
  BContact.setAttribute("data-align", "left");
  BContact.setAttribute("data-whatsapp", "21989710023");
  BContact.setAttribute("data-telegram", "marcuslorenlopes");
  BContact.setAttribute("data-callto", "21989710023");
  BContact.setAttribute("data-email", "marcusltorres@gmail.com");

  render(<App />, {
    container: document.body.appendChild(BContact),
  });

  const buttonElement = screen.getByTestId("button-bcontact");
  userEvent.click(buttonElement);
  const containerContact = await screen.findByTestId("container-contacts");

  expect(containerContact).toBeInTheDocument();
  expect(await screen.findByText("Whatsapp")).toBeInTheDocument();
  expect(await screen.findByText("Telegram")).toBeInTheDocument();
  expect(await screen.findByText("Email")).toBeInTheDocument();
  expect(await screen.findByText("Ligar")).toBeInTheDocument();

  // expect(containerContact).toMatchSnapshot();
});

// deve retornar um elemento ao adicionar apenas um atributo na tag html
test("should return an element when adding only one attribute in html tag", async () => {
  const BContact = document.createElement("div");
  BContact.setAttribute("id", "BContact");
  BContact.setAttribute("data-whatsapp", "21989710023");

  render(<App />, {
    container: document.body.appendChild(BContact),
  });

  const buttonElement = screen.getByTestId("button-bcontact");
  userEvent.click(buttonElement);

  expect(screen.queryByText("Whatsapp")).toBeInTheDocument();
  expect(screen.queryByText("Telegram")).not.toBeInTheDocument();
  expect(screen.queryByText("Email")).not.toBeInTheDocument();
  expect(screen.queryByText("Ligar")).not.toBeInTheDocument();
});
