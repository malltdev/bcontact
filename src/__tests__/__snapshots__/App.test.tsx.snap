// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return when clicking on the button a container with the list of contacts 1`] = `
<div
  class="sc-hKMtZM eFmTwW"
  data-testid="container-contacts"
>
  <div
    class="sc-eCYdqJ itnBEX"
  >
    <li>
      <a
        href="https://wa.me/21989710023"
        rel="noreferrer"
        target="_blank"
        title="Whatsapp"
      >
        <i>
          <img
            alt="Icone Whatsapp"
            src="https://raw.githubusercontent.com/malltdev/bcontact/main/lib/icons/whatsapp.svg"
          />
        </i>
        Whatsapp
      </a>
    </li>
    <li
      data-testid="list-contact"
    >
      <a
        href="https://t.me/marcuslorenlopes"
        rel="noreferrer"
        target="_blank"
        title="Telegram"
      >
        <i>
          <img
            alt="Icone Telegram"
            src="https://raw.githubusercontent.com/malltdev/bcontact/main/lib/icons/telegram.svg"
          />
        </i>
        Telegram
      </a>
    </li>
    <li
      data-testid="list-contact"
    >
      <a
        href="mailto:marcusltorres@gmail.com"
        title="Email"
      >
        <i>
          <img
            alt="Icone Email"
            src="https://raw.githubusercontent.com/malltdev/bcontact/main/lib/icons/email.svg"
          />
        </i>
        Email
      </a>
    </li>
    <li
      data-testid="list-contact"
    >
      <a
        href="tel:+21989710023"
        title="Ligar"
      >
        <i>
          <img
            alt="Icone Ligar"
            src="https://raw.githubusercontent.com/malltdev/bcontact/main/lib/icons/ligar.svg"
          />
        </i>
        Ligar
      </a>
    </li>
  </div>
  <div
    class="sc-gsnTZi dIYHNY"
  >
    <p>
      Powered by
       
      <strong>
        <a
          href="https://bcontact.com.br"
        >
          B.Contact
        </a>
      </strong>
    </p>
  </div>
</div>
`;
